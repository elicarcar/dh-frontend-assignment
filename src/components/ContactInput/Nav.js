import styled from "styled-components";

const Nav = styled.nav`
  position: relative;
  visibility: visible;
  opacity: 1;
  transition: visibility 0s, opacity 0.5s;
  transition-delay: 0.1s;
`;

export default Nav;
