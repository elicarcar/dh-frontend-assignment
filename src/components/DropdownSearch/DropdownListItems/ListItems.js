import styled from "styled-components";

const ListItems = styled.li.attrs(props => ({
  onClick: props.handleClick,
  displayItem: props.displayItem,
  id: props.id
}))`
  font-family: "Helvetica";
  color: #798697;
  line-height: 30px;
  padding: 1px;
  padding-left: 11px;
  font-size: 14px;
  cursor: ${props => (props.noMatch ? "not-allowed" : "pointer")};
  &:hover {
    color: #4a4a4a;
    background: #f7f7f7;
  }
`;

export default ListItems;
