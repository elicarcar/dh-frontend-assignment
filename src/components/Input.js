import styled from "styled-components";

const Input = styled.input.attrs(props => ({
  value: props.value,
  onChange: props.searchInput,
  onFocus: props.handleFocus,
  onBlur: props.handleBlur
}))`
  border: 1px solid #bfc5cd;
  border-radius: 7px;
  color: #798697;
  padding: 10px;
  outline: none;

  &:hover {
    border: 1px solid #4a4a4a;
  }
`;

export default Input;
