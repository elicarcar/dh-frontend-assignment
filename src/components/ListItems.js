import styled from "styled-components";

const ListItems = styled.li.attrs(props => ({
  onClick: props.setValue,
  onMouseUp: props.handleMouseUp,
  displayItem: props.displayItem,
  id: props.id
}))`
  color: #798697;
  line-height: 30px;
  display: ${props => (props.displayItem ? "block" : "none")}
  font-size: 14px;
  padding: 5px;
  cursor:pointer;
  &:hover {
    color: #4a4a4a;
    background: #f7f7f7;
  }
`;

export default ListItems;
