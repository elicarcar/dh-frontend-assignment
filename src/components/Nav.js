import styled from "styled-components";

const Nav = styled.nav.attrs(props => ({
  displayList: props.displayList
}))`
  display: ${props => (props.displayList ? "block" : "none")};
`;

export default Nav;
