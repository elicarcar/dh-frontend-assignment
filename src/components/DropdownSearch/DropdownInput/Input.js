import styled from "styled-components";
import Nav from "../DropdownSearchNav/Nav.js";
import Label from "../DropdownLabel/DropdownLabel";

const Input = styled.input.attrs(props => ({
  id: "contact-input",
  name: "contact",
  type: "text",
  placeholder: "Type or Search..",
  autoComplete: "off",
  value: props.value,
  onChange: props.handleInputChange,
  onBlur: props.handleBlur,
  onFocus: props.handleFocus,
  ref: props.inputRef
}))`
  border: 1px solid #bfc5cd;
  border-radius: 7px;
  color: #4a4a4a;
  padding: 10px;
  outline: none;
  transition: all 0.6s;
  transition-delay: 0.2s;

  &::-webkit-input-placeholder {
    opacity: 0;
  }

  &:hover {
    border: 1px solid #4a4a4a;
  }

  &:focus {
    text-indent: 12px;
  }

  &:focus::-webkit-input-placeholder {
    opacity: 1;
  }

  &:not(:focus) ~ ${Nav} {
    visibility: hidden;
    opacity: 0;
  }

  &:focus,
  &:not(:placeholder-shown) {
    & ~ ${Label} {
      transform: scale(0.9) translateY(-135%) translateX(-10px);
      opacity: 1;
      padding-bottom: 2px;
    }
  }
`;

export default Input;
