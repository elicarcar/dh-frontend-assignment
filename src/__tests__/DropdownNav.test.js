import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import React from "react";
import Input from "../components/DropdownSearch/DropdownInput/Input";
import Nav from "../components/DropdownSearch/DropdownSearchNav/Nav";
import ListItems from "../components/DropdownSearch/DropdownListItems/ListItems";
import "jest-styled-components";
import "@testing-library/jest-dom/extend-expect";

describe("<Input/>", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("<ListItems/>", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<ListItems />).toJSON();
    expect(tree).toHaveStyleRule("color", "#798697");
    expect(tree).toHaveStyleRule("line-height", "30px");
    expect(tree).toHaveStyleRule("padding", "1px");
    expect(tree).toHaveStyleRule("padding-left", "11px");
    expect(tree).toHaveStyleRule("font-size", "14px");
    expect(tree).toHaveStyleRule("color", "#4a4a4a", {
      modifier: ":hover"
    });
    expect(tree).toMatchSnapshot();
  });
});

describe("<Nav/>", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Nav />).toJSON();
    expect(tree).toHaveStyleRule("position", "relative");
    expect(tree).toHaveStyleRule("visibility", "visible");
    expect(tree).toHaveStyleRule("opacity", "1");
    expect(tree).toHaveStyleRule("transition", "visibility 0s,opacity 0.5s");
    expect(tree).toHaveStyleRule("transition-delay", "0.1s");
    expect(tree).toMatchSnapshot();
  });
});
