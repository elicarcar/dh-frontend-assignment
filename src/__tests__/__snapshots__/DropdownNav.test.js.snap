// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> matches the snapshot 1`] = `
.c0 {
  border: 1px solid #bfc5cd;
  border-radius: 7px;
  color: #4a4a4a;
  padding: 10px;
  outline: none;
  -webkit-transition: all 0.6s;
  transition: all 0.6s;
  -webkit-transition-delay: 0.2s;
  transition-delay: 0.2s;
}

.c0::-webkit-input-placeholder {
  opacity: 0;
}

.c0:hover {
  border: 1px solid #4a4a4a;
}

.c0:focus {
  text-indent: 12px;
}

.c0:focus::-webkit-input-placeholder {
  opacity: 1;
}

.c0:not(:focus) ~ .c1 {
  visibility: hidden;
  opacity: 0;
}

.c0:focus ~ .c2,
.c0:not(:placeholder-shown) ~ .c2 {
  -webkit-transform: scale(0.9) translateY(-135%) translateX(-10px);
  -ms-transform: scale(0.9) translateY(-135%) translateX(-10px);
  transform: scale(0.9) translateY(-135%) translateX(-10px);
  opacity: 1;
  padding-bottom: 2px;
}

<input
  autoComplete="off"
  className="c0"
  id="contact-input"
  name="contact"
  placeholder="Type or Search.."
  type="text"
/>
`;

exports[`<ListItems/> matches the snapshot 1`] = `
.c0 {
  font-family: "Helvetica";
  color: #798697;
  line-height: 30px;
  padding: 1px;
  padding-left: 11px;
  font-size: 14px;
  cursor: pointer;
}

.c0:hover {
  color: #4a4a4a;
  background: #f7f7f7;
}

<li
  className="c0"
/>
`;

exports[`<Nav/> matches the snapshot 1`] = `
.c1 {
  position: relative;
  visibility: visible;
  opacity: 1;
  -webkit-transition: visibility 0s,opacity 0.5s;
  transition: visibility 0s,opacity 0.5s;
  -webkit-transition-delay: 0.1s;
  transition-delay: 0.1s;
}

.c2:not(:focus) ~ .c0 {
  visibility: hidden;
  opacity: 0;
}

<nav
  className="c0 c1"
/>
`;
